packages/fx-core/src/plugins/resource/bot/plugin.ts

```diff
@@ -220,15 +224,21 @@ export class TeamsBotImpl {
     // 1. Do bot registration.
     if (this.config.scaffold.wayToRegisterBot === WayToRegisterBot.CreateNew) {
       await handler?.next(ProgressBarConstants.PROVISION_STEP_BOT_REG);
-      await this.createNewBotRegistrationOnAzure();
+      const botAuthCreds = await this.createOrGetBotAppRegistration();
+
+      if (!isArmSupportEnabled()) {
```

Comment:
the provision function will be skipped when arm enabled. There should be no arm enabled condition in it.

Reply:
Why? The provision function will not be skipped.

Reply:
Provision function is logically used to provision azure resource and handle related work based on the returned result of  provision. 
When arm is enabled, it will not provision azure resource. Related work also has no returned provision result to go on.

Reply:
Bot plugin needs to provision bot aad app so we need to save the aad registration logic in provision stage.